src/types.h

```diff
@@ -0,0 +1,84 @@
+#ifndef FIELD_TYPE_H
+#define FIELD_TYPE_H
+
+#include <napi.h>
+
+#include <arrow/type_fwd.h>
+
+// Don't use these macros outside of this file
+// They assume the existense of a Napi::Env env variable
+#define ENUMSET(name, enum)   Set(name, Napi::Number::New(env, static_cast<double>(enum)))
+#define F_ENUMSET(name, enum) type.ENUMSET(name, arrow::Type::type::enum)
+#define T_ENUMSET(name, enum) timeUnit.ENUMSET(name, arrow::TimeUnit::type::enum)
+
+// Enums that encompass all arrow types.
+
+// Types not supported by the parquet format or duplicated by other
+// types in the parquet format have been commented out, but left
+// for completeness in case that ever changes.
+
+// See FieldToNode() in <parquet/arrow/schema.cc> for the rational
+// behind these choices.
+namespace Types {
+  inline Napi::Object Init(Napi::Env env, Napi::Object exports) {
+    using Napi::Object;
+
+    Object type = Object::New(env);
+    F_ENUMSET("BOOL",                     BOOL),
+    F_ENUMSET("UINT8",                    UINT8),
+    F_ENUMSET("INT8",                     INT8),
+    F_ENUMSET("UINT16",                   UINT16),
+    F_ENUMSET("INT16",                    INT16),
+    F_ENUMSET("UINT32",                   UINT32),
+    F_ENUMSET("INT32",                    INT32),
+    // F_ENUMSET("UINT64",                   UINT64),                   // Not currently supported by Napi
+    F_ENUMSET("INT64",                    INT64),
+    // F_ENUMSET("HALF_FLOAT",               HALF_FLOAT),               // Not currently supported by parquet
+    F_ENUMSET("FLOAT",                    FLOAT),
+    F_ENUMSET("DOUBLE",                   DOUBLE),
+    F_ENUMSET("STRING",                   STRING),
+    // F_ENUMSET("BINARY",                   BINARY),                   // Not currently support by Napi
```

Comment:
`Buffer`?

Reply:
Oh, I'll look into that!

Reply:
it's a nodejs specific thing, not V8: https://stackoverflow.com/a/28952775/6303229